src/aqua/installation-spell/src/aqua/upload.aqua

```diff
@@ -30,23 +30,39 @@ func upload_worker_config(ipfs: Multiaddr, config: LocalWorkerConfig) -> CID:
     cid <- IpfsClient.upload_string(ipfs, json)
     <- cid
 
-func upload(config: LocalAppDeployConfig) -> AppDeployDefinition:
+-- worker definition uploading for workers deploy
+func upload_workers(config: LocalAppDeployConfig) -> AppDeployDefinition:
     deploy_defs: *WorkerDeployDefinition
-    ipfs_multiaddr: ?Multiaddr
 
-    if config.direct_deploy:
-        on HOST_PEER_ID:
-            res <- Ipfs.get_external_api_multiaddr()
+    for w <- config.workers:
+        definition_cid: ?CID
+        for h <- w.hosts:
+            on h via HOST_PEER_ID:
+                ipfs <- Ipfs.get_external_api_multiaddr()
+                -- cids should be the same for all hosts
+                -- but it definitely can be done better
+                definition_cid <- upload_worker_config(ipfs.multiaddr, w.config)
```

Comment:
maybe gather urls, do a unique over them, ans then upload a single time for each url?